### ISSUE TITLE
Support \tag* for marks

### ISSUE THREAD

**Issue by dginev** [brucemiller/LaTeXML#1953]
The reason I am opening this issue is that I saw this technique [recommended online](https://twitter.com/SamuelGWalters/status/1574508456039841792), and noticed latexml can't emulate it. Definitely not a high priority, but good to keep a record.

Minimal self-contained example:
```tex
\documentclass{article}
\usepackage{amsmath}
\usepackage{amsthm}
\begin{document}

\begin{proof}
\[
1+1=2  \tag*{\qed}
\]
done. \phantom{\qedhere}
\end{proof}

\end{document}
```

We would expect the QED mark at the usual spot for an equation number. Here's the PDF rendering:
<img width="700" src="https://user-images.githubusercontent.com/348975/192402808-77c5e284-920d-4dab-ad22-c805e4805215.png">

**Comment by brucemiller** [comment]
hmm... it used to work...

**Comment by dginev** [comment]
Oh, we should make it a test if so, might be a particularly fragile bit of logic.

**Comment by brucemiller** [comment]
Actually, it still works fine for numbered equations; but it apparently 
doesn't know that it's supposed to turn unnumbered one into numbered 
(sorta) ones.

**Comment by dginev** [comment]
Right, I had assumed we simply hadn't extended the support for somehow embedding arbitrary marks. Switching back to a "fresh bug report" label :>
